pkg/sdk/testint/users_integration_test.go

```diff
@@ -161,68 +257,690 @@ func TestInt_UserCreate(t *testing.T) {
+		id := testClientHelper().Ids.RandomAccountObjectIdentifier()
+
+		err := client.Users.Create(ctx, id, nil)
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().User.DropUserFunc(t, id))
+
+		assertDefaultParameters(id)
+	})
+
+	t.Run("alter: rename", func(t *testing.T) {
+		user, userCleanup := testClientHelper().User.CreateUser(t)
+		t.Cleanup(userCleanup)
+
+		newID := testClientHelper().Ids.RandomAccountObjectIdentifier()
+		alterOptions := &sdk.AlterUserOptions{
+			NewName: newID,
+		}
+		err := client.Users.Alter(ctx, user.ID(), alterOptions)
+		require.NoError(t, err)
+		t.Cleanup(testClientHelper().User.DropUserFunc(t, newID))
+
+		result, err := client.Users.Describe(ctx, newID)
+		require.NoError(t, err)
+		assert.Equal(t, newID.Name(), result.Name.Value)
+	})
+
+	t.Run("alter: set and unset", func(t *testing.T) {
+		user, userCleanup := testClientHelper().User.CreateUser(t)
+		t.Cleanup(userCleanup)
+
+		currentRole := testClientHelper().Context.CurrentRole(t)
+
+		objectAssert.AssertThatObject(t, objectAssert.UserFromObject(t, user).
+			HasDefaults(user.Name).
+			HasDisplayName(user.Name).
+			HasOwner(currentRole.Name()),
+		)
+
+		alterOpts := &sdk.AlterUserOptions{Set: &sdk.UserSet{
+			ObjectProperties: &sdk.UserObjectProperties{
+				Password:           sdk.String(password),
+				LoginName:          sdk.String(newValue),
+				DisplayName:        sdk.String(newValue),
+				FirstName:          sdk.String(newValue),
+				MiddleName:         sdk.String(newValue),
+				LastName:           sdk.String(newValue),
+				Email:              sdk.String(email),
+				MustChangePassword: sdk.Bool(true),
+				Disable:            sdk.Bool(true),
+				DaysToExpiry:       sdk.Int(5),
+				MinsToUnlock:       sdk.Int(15),
+				DefaultWarehouse:   sdk.Pointer(warehouseId),
+				DefaultNamespace:   sdk.Pointer(schemaIdObjectIdentifier),
+				DefaultRole:        sdk.Pointer(roleId),
+				DefaultSecondaryRoles: &sdk.SecondaryRoles{
+					Roles: []sdk.SecondaryRole{{Value: "ALL"}},
+				},
+				MinsToBypassMFA: sdk.Int(30),
+				RSAPublicKey:    sdk.String(key),
+				RSAPublicKey2:   sdk.String(key2),
+				Comment:         sdk.String("some comment"),
+			},
+		}}
+
+		err := client.Users.Alter(ctx, user.ID(), alterOpts)
+		require.NoError(t, err)
+
+		objectAssert.AssertThatObject(t, objectAssert.User(t, user.ID()).
+			HasName(user.Name).
+			HasCreatedOnNotEmpty().
+			// login name is always case-insensitive
+			HasLoginName(strings.ToUpper(newValue)).
+			HasDisplayName(newValue).
+			HasFirstName(newValue).
+			HasLastName(newValue).
+			HasEmail(email).
+			HasMinsToUnlock("14").
+			HasDaysToExpiryNotEmpty().
+			HasComment("some comment").
+			HasDisabled(true).
+			HasMustChangePassword(true).
+			HasSnowflakeLock(false).
+			HasDefaultWarehouse(warehouseId.Name()).
+			HasDefaultNamespaceId(schemaId).
+			HasDefaultRole(roleId.Name()).
+			HasDefaultSecondaryRoles(`["ALL"]`).
```

Comment:
Are we planning to add sdk-level function to parse this field?

Reply:
TBH I don't know yet. Currently, as I understand the docs, only ["ALL"] value is supported, therefore I think we can just leave it with a TODO when more options are supported.